core/includes/install.core.inc

```diff
@@ -543,13 +553,33 @@ function install_tasks($install_state) {
   }
   $display_select_profile = $hidden_profile_count > 1 && count($install_state['profiles']) != 1;
 
+  // Determine whether a translation file must be downloaded during the
+  // 'install_download_translation' task. Download when a non-English language
+  // is selected, but no translation is yet in the translations directory.
+  $needs_download = FALSE;
+  if (isset($install_state['parameters']['langcode']) && $install_state['parameters']['langcode'] != 'en') {
+    if (isset($install_state['translations'])) {
+      $local_file_found = FALSE;
+      foreach($install_state['translations'] as $translation_file) {
+        if ($translation_file->langcode == $install_state['parameters']['langcode']) {
+          $local_file_found = TRUE;
+          break;
+        }
+      }
+      $needs_download = !$local_file_found;
+    }
+  }
+
   // Start with the core installation tasks that run before handing control
   // to the installation profile.
   $tasks = array(
     'install_select_language' => array(
       'display_name' => st('Choose language'),
       'run' => INSTALL_TASK_RUN_IF_REACHED,
     ),
+    'install_download_translation' => [
+      'run' => $needs_download ? INSTALL_TASK_RUN_IF_REACHED : INSTALL_TASK_SKIP,
+    ],
```

Comment:
I'm happy to switch to the shorthand array format, but after we've made a policy decision about it. For now, let keep things consistent with the rest of the code here please:
```suggestion
    'install_download_translation' => array(
      'run' => $needs_download ? INSTALL_TASK_RUN_IF_REACHED : INSTALL_TASK_SKIP,
    ),
```

Reply:
I agree with this particular chunk of code, since it should best be consistent with the rest of the array. But what's our policy on having the bracket syntax mixed *in the same file* though? There's a bit more of it lower down. I'm happy to change all of it to the old syntax but curious what others might think. Leave it, change it, mix it?

Reply:
Done